uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitBasicAttributeView.java

```diff
@@ -71,24 +71,30 @@ private BasicFileAttributes buildAttrs(final JGitFileSystem fs,
 
         final Ref ref = fs.getGit().getRef(branchName);
 
-        final List<Date> records = new ArrayList<>();
+        final List<RevCommit> records = new ArrayList<>();
+        final Long[] result = new Long[2];
+
 
         if (ref != null) {
             try {
                 final CommitHistory history = fs.getGit().listCommits(ref, pathInfo.getPath());
-                for (final RevCommit commit : history.getCommits()) {
-                    records.add(commit.getAuthorIdent().getWhen());
-                }
+                List<RevCommit> rec  = history.getCommits();
+                result[0]  = rec.get(0).getAuthorIdent().getWhen().getTime();
+                result[1]  = rec.get(rec.size() - 1).getAuthorIdent().getWhen().getTime();
+
+//                for (final RevCommit commit : history.getCommits()) {
+//                    records.add(commit.getAuthorIdent().getWhen());
+//                }
             } catch (Exception e) {
```

Comment:
@RishiRajAnand I tested this change with a fairly large niogit repo, and could not see a performance improvement. Upon further testing, I realized this costly lookup of the jgit history is happening too frequently at this location, so I changed this to a lazy approach whenever one of the lastModifiedTime() or creationTime() methods are called, see PR#1132 for details. With this change, performance is back to where it was before the RHPAM-2865 change. Please let me know what you think.

Reply:
@martinweiler I commented a few Stuff in PR#1132 that I think we need to change (very small stuff) and if the performance is better we should ask QE to review and confirm. Good job!

Reply:
@adrielparedes thanks, I have provided the numbers from my local testing in the other PR.